### PR TITLE
fix: make `proof_wanted` not report unused variables

### LINKED ISSUE #938: `proof_wanted` is linted for unused arguments
We should disable the unused argument linter and the new variable inclusion mechanism linter in `proof_wanted`. Currently, we have the following behavior:
```
import Batteries.Util.ProofWanted

class Foo where

variable [Foo]

proof_wanted two_eq_two (h : 1 + 1 = 2) : 2 = 2
/-
unused variable `h`
note: this linter can be disabled with `set_option linter.unusedVariables false`

included section variable '[Foo]' is not used in 'two_eq_two', consider excluding it
-/
```

### PR DESCRIPTION
**Pull request by kmill**
This was requested by Terence Tao [on Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Is.20a.20.60definition_wanted.60.20keyword.20possible.3F/near/477325788)

Closes #938

### CODE REVIEW

File: Batteries/Util/ProofWanted.lean [hunk 1]
```diff
@@ -33,7 +33,10 @@ elaboration, but it's then removed from the environment.
...
     -- The helper axiom is used instead of `sorry` to avoid spurious warnings
-    elabCommand <| ← `(axiom helper (p : Prop) : p
-                       $mods:declModifiers
-                       theorem $name $args* : $res := helper _)
+    elabCommand <| ← `(
+      section
+      set_option linter.unusedVariables false
+      axiom helper {α : Sort _} : α
+      $mods:declModifiers theorem $name $args* : $res := helper
+      end)
```
Comment (kmill) [inline Batteries/Util/ProofWanted.lean:41]: Another solution would be to use something like [Gabriel's `sorry`](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Is.20a.20.60definition_wanted.60.20keyword.20possible.3F/near/477287818) and capture all the fvars, but that would take making a term elaborator to replace `helper`.

File: Batteries/Util/ProofWanted.lean [hunk 2]
```diff
@@ -33,7 +33,10 @@ elaboration, but it's then removed from the environment.
 def elabProofWanted : CommandElab
   | `($mods:declModifiers proof_wanted $name $args* : $res) => withoutModifyingEnv do
     -- The helper axiom is used instead of `sorry` to avoid spurious warnings
-    elabCommand <| ← `(axiom helper (p : Prop) : p
-                       $mods:declModifiers
-                       theorem $name $args* : $res := helper _)
+    elabCommand <| ← `(
+      section
+      set_option linter.unusedVariables false
+      axiom helper {α : Sort _} : α
```
Comment (fgdorais) [inline Batteries/Util/ProofWanted.lean:39]: Do we really want `Sort _` here?

Comment (fgdorais) [reply Batteries/Util/ProofWanted.lean:39]: Ah, I see this incorporates `definition_wanted`.

Comment (kmill) [reply Batteries/Util/ProofWanted.lean:39]: Yes, I found that the error messages for non-theorems was unfriendly for just Prop. The test file shows that non-Prop still gives a good error.

File: Batteries/Util/ProofWanted.lean (hunk 1 above)
Comment (fgdorais) [reply Batteries/Util/ProofWanted.lean:41]: That would be something upstream.

**Pull request merged**